tests/ha/upgrade_from_sle11sp4_workarounds.pm

```diff
@@ -29,6 +29,35 @@ sub run {
     # We execute this test after a reboot, so we need to log in
     select_console 'root-console';
 
+    # In 2-node migration from SLES+HA 11-SP4, host name resolution
+    # is needed earlier than in other test cases to apply the workarounds
+    # needed by the cluster after migration. We'll add the hosts
+    # to /etc/hosts if name resolution is failing
+    if (is_node(1) or is_node(2)) {
+        my $hostname = get_hostname;
+        my $partner  = $hostname;
+        $partner =~ s/node([0-9]+)$/node/;
+        $partner = $1 eq '01' ? $partner . "02" : $partner . "01";
```

Comment:
Maybe you can have a look at the `choose_node` function available in `lib/hacluster.pm` and check if you can use it here.

Reply:
Will use `choose_node`. Thanks